// library/CustomerDetailRow.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');
const Label = require('sf-core/ui/label');
const Font = require('sf-core/ui/font');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const CustomerDetailRow = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		const flCustomerDetailLineStyle = getCombinedStyle(".flexLayout", {
			width: null,
			height: 0.5,
			backgroundColor: Color.create(255, 95, 198, 144),
			bottom: 0,
			left: 0,
			right: 0,
			positionType: FlexLayout.PositionType.ABSOLUTE
		});  
		var flCustomerDetailLine = new FlexLayout(flCustomerDetailLineStyle);  
		this.addChild(flCustomerDetailLine);
		this.flCustomerDetailLine = flCustomerDetailLine;

		const lblFieldNameStyle = getCombinedStyle(".label", {
			height: 14,
			width: null,
			backgroundColor: Color.create(0, 255, 255, 255),
			textColor: Color.create(255, 29, 29, 38),
			text: "Customer Number",
			alpha: 0.51,
			marginLeft: 0.5,
			font: Font.create("Lato", 12, Font.NORMAL)
		});  
		var lblFieldName = new Label(lblFieldNameStyle);  
		if(lblFieldNameStyle.scrollEnabled === false)
			lblFieldName.ios && (lblFieldName.ios.scrollEnabled = false);
		this.addChild(lblFieldName);
		this.lblFieldName = lblFieldName;

		const lblFieldValueStyle = getCombinedStyle(".label", {
			width: null,
			height: 16,
			marginLeft: 0.5,
			textColor: Color.create(255, 29, 29, 38),
			marginTop: 7,
			text: "7645980",
			font: Font.create("Lato", 14, Font.NORMAL)
		});  
		var lblFieldValue = new Label(lblFieldValueStyle);  
		if(lblFieldValueStyle.scrollEnabled === false)
			lblFieldValue.ios && (lblFieldValue.ios.scrollEnabled = false);
		this.addChild(lblFieldValue);
		this.lblFieldValue = lblFieldValue;

		//assign the children to customerDetailRow 
		this.children = Object.assign({}, {
			flCustomerDetailLine: flCustomerDetailLine,
			lblFieldName: lblFieldName,
			lblFieldValue: lblFieldValue
		});
		
	});

CustomerDetailRow.defaults = getCombinedStyle(".flexLayout", {
	width: null,
	height: 61.75,
	alpha: 1,
	paddingTop: 12.75
});

module && (module.exports = CustomerDetailRow);
