// library/DialogWait.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');
const ActivityIndicator = require('sf-core/ui/activityindicator');


const DialogWait = extend(FlexLayout)(
	//constructor
	function(_super, props){
		// initalizes super class for this component scope
		_super(this, props);


		var flWaitWhite = new FlexLayout({
			height: 150,
			alignContent: FlexLayout.AlignContent.STRETCH,
			alignItems: FlexLayout.AlignItems.CENTER,
			justifyContent: FlexLayout.JustifyContent.CENTER,
			flexWrap: FlexLayout.FlexWrap.NOWRAP,
			flexDirection: FlexLayout.FlexDirection.COLUMN,
			positionType: FlexLayout.PositionType.RELATIVE,
			marginLeft: 60,
			marginRight: 60,
			backgroundColor: Color.create("#FFFFFF"),
			alpha: 1,
			borderColor: Color.create(255, 255, 255, 255),
			borderWidth: 1,
			borderRadius: 15,
			visible: true
		}); 
		this.addChild(flWaitWhite);
		
		var aiWait = new ActivityIndicator({
			width: 42,
			height: 42,
			positionType: FlexLayout.PositionType.RELATIVE,
			backgroundColor: Color.create(0, 255, 255, 255),
			alpha: 1,
			borderColor: Color.create(255, 0, 0, 0),
			borderWidth: 0,
			visible: true,
			color: Color.create(255, 44, 50, 57)
		}); 
		flWaitWhite.addChild(aiWait);
		
		
		//assign the children to dialogWait 
		this.children = Object.assign({}, {
			flWaitWhite: flWaitWhite
		});
		
		//assign the children of flWaitWhite
		flWaitWhite.children =  Object.assign({}, {
			aiWait: aiWait
		});
	}
);

DialogWait.defaults = {
	left: 0,
	top: 0,
	alignContent: FlexLayout.AlignContent.STRETCH,
	alignItems: FlexLayout.AlignItems.STRETCH,
	justifyContent: FlexLayout.JustifyContent.CENTER,
	flexWrap: FlexLayout.FlexWrap.NOWRAP,
	flexDirection: FlexLayout.FlexDirection.COLUMN,
	positionType: FlexLayout.PositionType.ABSOLUTE,
	right: 0,
	bottom: 0,
	backgroundColor: Color.create(204, 0, 0, 0),
	alpha: 1,
	borderColor: Color.create(255, 0, 0, 0),
	borderWidth: 0,
	visible: true
 };

module && (module.exports = DialogWait);
